ci/scripts/wait-for-other-builds-to-finish.sh

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+: ${CIRCLECI_API_KEY:?"Required Env Variable not found!"}
+: ${CIRCLE_BUILD_NUM:?"Required Env Variable not found!"}
+: ${CIRCLE_PROJECT_USERNAME:?"Required Env Variable not found!"}
+: ${CIRCLE_PROJECT_REPONAME:?"Required Env Variable not found!"}
+: ${CIRCLE_REPOSITORY_URL:?"Required Env Variable not found!"}
+
+if [ -z "$1" ]; then
+	echo "Must provide Max Queue Time in *minutes* as script argument"
+	exit 1
+fi
+
+readonly MAX_TIME=$1
+
+wait_for_previous_builds_to_complete(){
+	local max_time_seconds=$((MAX_TIME * 60))
+
+	until [[ $(curl -s "https://circleci.com/api/v1.1/project/github/${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}?circle-token=${CIRCLECI_API_KEY}&filter=running" \
```

Comment:
So, locally:
```
Using build environment variables:
  BASH_ENV=/tmp/.bash_env-localbuild-1533734475
  CI=true
  CIRCLECI=true
  CIRCLE_BRANCH=oh-bosh
  CIRCLE_BUILD_NUM=
  CIRCLE_JOB=build
  CIRCLE_NODE_INDEX=0
  CIRCLE_NODE_TOTAL=1
```
`CIRCLE_BUILD_NUM` is not set, we should make the script noop if that is the case